sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala

```diff
@@ -112,18 +115,18 @@ private[hive] object IsolatedClientLoader extends Logging {
   private def downloadVersion(
       version: HiveVersion,
       hadoopVersion: String,
-      ivyPath: Option[String]): Seq[URL] = {
+      ivyPath: Option[String],
+      centralRepo: String): Seq[URL] = {
     val hiveArtifacts = version.extraDeps ++
       Seq("hive-metastore", "hive-exec", "hive-common", "hive-serde")
         .map(a => s"org.apache.hive:$a:${version.fullVersion}") ++
       Seq("com.google.guava:guava:14.0.1",
         s"org.apache.hadoop:hadoop-client:$hadoopVersion")
-
     val classpath = quietly {
       SparkSubmitUtils.resolveMavenCoordinates(
         hiveArtifacts.mkString(","),
         SparkSubmitUtils.buildIvySettings(
-          Some("http://www.datanucleus.org/downloads/maven2"),
```

Comment:
Interesting. So, with this PR, the side-effect benefit is the removal of the flakiness by default, @xuanyuanking?

Reply:
@xuanyuanking . If then, could you make a separate PR for this line change? Then, we can backport your new PR to branch-2.4, too. After that, we can proceed this PR on top of that.

Reply:
Yes, as the discussion above, the flakiness is caused by when the Jenkins blocked by maven central repo and the additional datanucleus remote repo still not work.
I updated this PR to set google mirror as an additional remote repo for hive tests in 49ea1cd.

Reply:
No. What I meant was another PR like the following two lines (excluding all the other stuff in this PR) for `master` and `branch-2.4`.
```
-           Some("http://www.datanucleus.org/downloads/maven2"),
+          Some(""),
```

Reply:
Maybe, not this line. `TestHive.scala` will be a better place. But, we should not have `private[spark] val ADDITIONAL_REMOTE_REPOSITORIES =...` in that PR.